modules/github-events/README.md

```diff
@@ -0,0 +1,102 @@
+# `github-events`
+
+This module provisions infrastructure to listen to webhook events from GitHub and
+publish them to a broker.
+
+More information on GitHub webhooks:
+- https://docs.github.com/en/webhooks/using-webhooks/validating-webhook-deliveries
+- https://docs.github.com/en/webhooks/webhook-events-and-payloads
+
+Events are published as they are received from GitHub, and are not transformed in
+any way. The events are published to a broker, which can be used to fan out the
+events to other services, or filter based on their type.
+
+You can use this with `cloudevent-recorder` to record GitHub events to a BigQuery table.
+
+CloudEvent types are derived from the GitHub event type, and are prefixed with
+`dev.chainguard.github`. For example, the `push` event is published as
+`dev.chainguard.github.push`.
+
+```hcl
+// Create a network with several regional subnets
+module "networking" {
+  source = "chainguard-dev/common/infra//modules/networking"
+
+  name       = "my-networking"
+  project_id = var.project_id
+  regions    = [...]
+}
+
+// Create the Broker abstraction.
+module "cloudevent-broker" {
+  source = "chainguard-dev/common/infra//modules/cloudevent-broker"
+
+  name       = "my-broker"
+  project_id = var.project_id
+  regions    = module.networking.regional-networks
+}
+
+// Forward events to the broker.
+module "github-events" {
+  source = "./modules/github-events"
+
+  project_id = var.project_id
+  name       = "github-events"
+  regions    = module.networking.regional-networks
+  ingress    = module.cloudevent-broker.ingress
+
+  // Which user is allowed to populate webhook secret values.
+  secret_version_adder = "user:you@company.biz"
+}
+```
+
+After applying this, generate a random secret value and add it to the GitHub
```

Comment:
can we use github tf provider to create the webhook and populate the secret?

Reply:
Something like this https://github.com/chainguard-dev/infra-images/pull/648/files#diff-8789f1c464f781d4f07e97ec136d38681e1b4c20108a5eaaa5fbfe7f110efbfe

Reply:
Yeah, that could be done by the caller of this module. Maybe the module should take an input of the webhook secret, which it could populate the initial version of the secret with.

I'm a little uneasy about TF doing secret rotation, since it would need to coordinate between both places and any time in between we'd drop messages.